code/userprog/addrspace.cc

```diff
@@ -149,7 +149,7 @@ AddrSpace::AddrSpace(OpenFile *executable) : fileTable(MaxOpenFiles) {
 					numPages, size);
 // first, set up the translation 
     pageTable = new TranslationEntry[numPages + (50 * 8)];
-	IPT = new TranslationEntry[numPages + (50*8)];
+	IPT = new IPTEntry[numPages + (50*8)]; //size will eventually be 32
```

Comment:
Should declare this outside addrspace since each IPTEntry will ahve a pointer to addrspace.h owner